### PR TITLE
txn: add more debug information for the txn commands

### LINKED ISSUE #12803: txn: enhance the debug information
## Development Task
In https://github.com/pingcap/tidb/issues/34875 it takes quite a long time to investigate, the debug log information helps a lot, but there's still some important information missing such as retrying or 1pc flags.  Try to enhance the information to reduce the investigation cost.

### PR DESCRIPTION
**Pull request by cfzjywxk**
Signed-off-by: cfzjywxk <lsswxrxr@163.com>

### What is changed and how it works?

Issue Number: close #12803 

What's Changed:

```commit-message
In the investigation process, the debug log information helps a lot, but there's still some important information 
missing such as retrying or 1pc flags. 
Changed:
1. Add more useful information displaying commands.
2. Redact necessary parts.
```

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)

Side effects

### Release note 

```release-note
NONE
```

### CODE REVIEW

**Comment by ekexium** [comment]
Shall we redact these logs?

**Comment by ekexium** [comment]
Seems some fields are not redacted when setting `redact-info-log=true`. Is it expected?
```log
[2022/06/08 12:12:35.422 +08:00] [DEBUG] [scheduler.rs:414] ["received new command"] [cmd="kv::command::prewrite mutations([Put((?, [8, 2, 2, 40, 116, 105, 107, 118, 95, 103, 99, 95, 108, 101, 97, 100, 101, 114, 95, 108, 101, 97, 115, 101, 8, 4, 2, 46, 50, 48, 50, 50, 48, 54, 48, 56, 45, 49, 50, 58, 49, 52, 58, 51, 53, 32, 43, 48, 56, 48, 48, 8, 6, 2, 90, 67, 117, 114, 114, 101, 110, 116, 32, 71, 67, 32, 119, 111, 114, 107, 101, 114, 32, 108, 101, 97, 100, 101, 114, 32, 108, 101, 97, 115, 101, 46, 32, 40, 68, 79, 32, 78, 79, 84, 32, 69, 68, 73, 84, 41]), None)]) primary([116, 128, 0, 0, 0, 0, 0, 0, 17, 95, 114, 128, 0, 0, 0, 0, 1, 102, 241]) secondary(None)@ 433759598777466883 3002 false 1 433759598777466884 0 false| region_id: 18 region_epoch { conf_ver: 1 version: 9 } peer { id: 19 store_id: 1 } max_execution_duration_ms: 20000 resource_group_tag: ? disk_full_opt: AllowedOnAlmostFull"] [cid=339]
```

**Comment by cfzjywxk** [comment]
> [2022/06/08 12:12:35.422 +08:00] [DEBUG] [scheduler.rs:414] ["received new command"] [cmd="kv::command::prewrite mutations([Put((?, [8, 2, 2, 40, 116, 105, 107, 118, 95, 103, 99, 95, 108, 101, 97, 100, 101, 114, 95, 108, 101, 97, 115, 101, 8, 4, 2, 46, 50, 48, 50, 50, 48, 54, 48, 56, 45, 49, 50, 58, 49, 52, 58, 51, 53, 32, 43, 48, 56, 48, 48, 8, 6, 2, 90, 67, 117, 114, 114, 101, 110, 116, 32, 71, 67, 32, 119, 111, 114, 107, 101, 114, 32, 108, 101, 97, 100, 101, 114, 32, 108, 101, 97, 115, 101, 46, 32, 40, 68, 79, 32, 78, 79, 84, 32, 69, 68, 73, 84, 41]), None)]) primary([116, 128, 0, 0, 0, 0, 0, 0, 17, 95, 114, 128, 0, 0, 0, 0, 1, 102, 241]) secondary(None)@ 433759598777466883 3002 false 1 433759598777466884 0 false| region_id: 18 region_epoch { conf_ver: 1 version: 9 } peer { id: 19 store_id: 1 } max_execution_duration_ms: 20000 resource_group_tag: ? disk_full_opt: AllowedOnAlmostFull"] [cid=339]

It's needed, seems the `value` in `Mutation` is not redacted before.

**Review by zhangjinpeng1987 — APPROVED** [review]
LGTM

**Comment by cfzjywxk** [comment]
@ekexium 
The non-redacted parts are processed accordingly, PTAL thx.

**Review by ekexium — APPROVED** [review]

File: src/storage/txn/commands/acquire_pessimistic_lock.rs [hunk 1]
```diff
@@ -26,7 +26,8 @@ command! {
     /// This can be rolled back with a [`PessimisticRollback`](Command::PessimisticRollback) command.
     AcquirePessimisticLock:
         cmd_ty => StorageResult<PessimisticLockRes>,
-        display => "kv::command::acquirepessimisticlock keys({}) @ {} {} | {:?}", (keys.len, start_ts, for_update_ts, ctx),
+        display => "kv::command::acquirepessimisticlock keys({:?}) @ {} {} {} {:?} {} {} | {:?}",
```
Comment (MyonKeminta) [inline src/storage/txn/commands/acquire_pessimistic_lock.rs:29]: I'm afraid it looks a little hard to recognize which field is each value in the logs.

**Review by MyonKeminta — APPROVED** [review]
Ah, I'm sorry. Just noticed that the `Key` type already has built-in redaction.

File: components/txn_types/src/types.rs [hunk 2]
```diff
@@ -284,6 +284,42 @@ pub enum Mutation {
...
+    }
+}
+
+impl Display for Mutation {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            Mutation::Put((key, value), assertion) => write!(
+                f,
+                "Put key:{:?} value:{:?} assertion:{:?}",
+                key,
```
[the remark this answers is not in the document]
Comment (ekexium) [reply components/txn_types/src/types.rs:299]: `Key` has implemented `Debug` and `Display` using log_wrappers

**Pull request merged**